### ISSUE TITLE
audio/libmad: checking size of int... configure: error: cannot compute sizeof (int), 77

### ISSUE THREAD

**Issue by fpusersuggest** [HardenedBSD/hardenedbsd-ports#152]
Hi,
I am trying to build xfburn, but it stops  building at the audio/libmad configure:

```
checking whether byte ordering is bigendian... no
checking for pid_t... (cached) yes
checking for int... (cached) yes
checking size of int... configure: error: cannot compute sizeof (int), 77
See `config.log' for more details.
===>  Script "configure" failed unexpectedly.
Please report the problem to sunpoet@FreeBSD.org [maintainer] and attach the
"/usr/ports/audio/libmad/work/libmad-0.15.1b/config.log" including the output
of the failure of your make command. Also, it might be a good idea to provide
an overview of all packages installed on your system (e.g. a
/usr/local/sbin/pkg-static info -g -Ea).
*** Error code 1

Stop.
make[1]: stopped in /usr/ports/audio/libmad
*** Error code 1

Stop.
```

it works only if I disable PIE, RELRO, SAFESTACK.
I'm on:
`FreeBSD mypcname 11.2-STABLE-HBSD FreeBSD 11.2-STABLE-HBSD #0 [STABLE:HardenedBSD-11-STABLE-v1100056]: Wed Jul  4 02:15:08 UTC 2018     root@nyi-01.build.hardenedbsd.org:/usr/obj/usr/src/sys/HARDENEDBSD  amd64`

**Comment by pkubaj** [comment]
Can you paste /usr/ports/audio/libmad/work/libmad-0.15.1b/config.log?

On 18-07-23 03:47:45, MysticDoor wrote:
>Hi,
>I am trying to build xfburn, but it stops  building at the audio/libmad configure:
>
>```
>checking whether byte ordering is bigendian... no
>checking for pid_t... (cached) yes
>checking for int... (cached) yes
>checking size of int... configure: error: cannot compute sizeof (int), 77
>See `config.log' for more details.
>===>  Script "configure" failed unexpectedly.
>Please report the problem to sunpoet@FreeBSD.org [maintainer] and attach the
>"/usr/ports/audio/libmad/work/libmad-0.15.1b/config.log" including the output
>of the failure of your make command. Also, it might be a good idea to provide
>an overview of all packages installed on your system (e.g. a
>/usr/local/sbin/pkg-static info -g -Ea).
>*** Error code 1
>
>Stop.
>make[1]: stopped in /usr/ports/audio/libmad
>*** Error code 1
>
>Stop.
>```
>
>it works only if I disable PIE, RELRO, SAFESTACK.
>I'm on:
>`FreeBSD mypcname 11.2-STABLE-HBSD FreeBSD 11.2-STABLE-HBSD #0 [STABLE:HardenedBSD-11-STABLE-v1100056]: Wed Jul  4 02:15:08 UTC 2018     root@nyi-01.build.hardenedbsd.org:/usr/obj/usr/src/sys/HARDENEDBSD  amd64`
>
>
>
>
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/HardenedBSD/hardenedbsd-ports/issues/152
>-- 
>This message has been scanned for viruses and
>dangerous content by MailScanner, and is
>believed to be clean.
>

-- 
 _________________________________________ 
/ I look at life as being cruise director \
| on the Titanic. I may not get there,    |
| but I'm going first class.              |
|                                         |
\ -- Art Buchwald                         /
 ----------------------------------------- 
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

**Comment by fpusersuggest** [comment]
Yeah I have enabled them manually... is there a way to enable HARDENING options globally ? I have dozens of dialogs to change..

**Comment by lattera** [comment]
Some of the hardening options should not be enabled globally. SafeStack is a good example. Additionally, if PIE and RELRO have been explicitly disabled for a port, there's likely good reason why.

Instead, users who want to toy with the hardening options should set them on a per-port basis. If an option is disabled for that port, but the user has found that option works for that port, a bug report should be filed to tell us that it's okay to enable that option for that port.
